android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt

```diff
@@ -135,54 +128,56 @@ class BackgroundService : AccessibilityService() {
       return
     }
 
-    val parentNodeInfo: AccessibilityNodeInfo = event.source ?: return
-
-    if (overlayIsActivated(applicationContext) && isWindowChangeEvent(event)) {
+    if (overlayIsActivated(applicationContext)) {
       val packageName = event.packageName.toString()
 
-      if (isLauncherPackage(packageName) && parentNodeInfo.className.toString() != "android.widget.FrameLayout") {
-        _hide = "true"
-        handler.post(runnableCode)
-        return
+      if (getEventType(event) == "TYPE_WINDOW_STATE_CHANGED" && packageName != "com.android.chrome") {
+        if (packageName.contains("com.google.android.inputmethod") ||
+          packageName == "com.google.android.googlequicksearchbox" ||
+          packageName == "com.android.systemui"
+        ) {
+          _packageName = packageName
+          _hide = "true"
+          handler.post(runnableCode)
+          return
+        }
       }
 
-      if (outsideChrome(parentNodeInfo)) {
+      if (isLauncherActivated(packageName)) {
```

Comment:
What if we switched to another application ?

Reply:
How would you do that?

Reply:
I think maybe it’s better to detect that we are **not** on the targeted applications, than trying to detect every possible case that we are not (launcher, non-targeted application, settings screen, …)

Reply:
Some app events can run in background so can't exactly do this

Reply:
Can’t we just check current active application before triggering a display ?

Reply:
I let you check this out